static/js/drafts.js

```diff
@@ -136,19 +148,71 @@ exports.restore_draft = function (draft_id) {
         }
     } else {
         if (draft.private_message_recipient !== "") {
-            narrow.activate([{operator: "pm-with", operand: draft.private_message_recipient}],
+            var recipients = draft.private_message_recipient.replace(/\s/g, '');
+            narrow.activate([{operator: "pm-with", operand: recipients}],
                              {select_first_unread: true, trigger: "restore draft"});
         }
     }
 
     overlays.close_overlay("drafts");
-    compose_fade.clear_compose();
-    if (draft.type === "stream" && draft.stream === "") {
-        draft_copy.subject = "";
+    load_restored_draft(draft_id);
+};
+
+exports.draft_matching_narrow = function (operators) {
+    if ($("#new_message_content").val()) {
+        return; // Abort if the user currently has something in the compose box
+    }
+
+    var filter = new Filter(operators);
+    // Only take from 2 most recent drafts to avoid intrusiveness
+    var draft_list = Object.keys(draft_model.get()).slice(-2);
+    var restore;
+    if (filter.has_operator('pm-with')) {
+        if (operators.length > 1) {
+            return;
+        }
+        draft_list = _.filter(draft_list, function (draft_id) {
+            if (draft_model.getDraft(draft_id).type === 'stream') {
```

Comment:
we should say `!== 'private'` here.  I would also consider extracting two simple functions `get_private_drafts` and `get_stream_drafts`